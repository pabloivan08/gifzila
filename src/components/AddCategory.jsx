import { useState } from "react"

const AddCategory = ({ onAddCategory }) => {

  const [ inputValue, setInputValue ] = useState('')

  const onInputChange = ( event ) => {
    setInputValue( event.target.value )
  }

  const onSubmit = ( event ) => {
    event.preventDefault()
    inputValue.trim() !== '' && onAddCategory( inputValue.trim() )
    setInputValue('')
  }

  return(
    <form onSubmit={ onSubmit }>
      <input 
        type="text" 
        placeholder="Search a cool Gif"
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  )
}

export { AddCategory }