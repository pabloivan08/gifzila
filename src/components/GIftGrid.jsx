import { GifItem } from "./GifItem"
import { useFetchGifs } from "../hooks/useFetchGifs"

const GiftGrid = ({ category }) => {

  const {images, isLoading} = useFetchGifs( category )

  return(
    <>
    <h3>{ category }</h3>

    { isLoading && <h4>Cargando...</h4> }

    <div className="card-grid"> 
    {
      images.map( gif => <GifItem key={ gif.id } { ...gif } /> )
    }
    </div>
    </>
  )
}

export { GiftGrid }