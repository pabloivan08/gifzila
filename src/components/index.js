export * from './AddCategory'
export * from './GifItem'
export * from './GIftGrid'