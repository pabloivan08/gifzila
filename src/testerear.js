// Print in console 'hello world'
