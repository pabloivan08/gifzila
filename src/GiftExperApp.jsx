import { useState } from "react"
import { AddCategory, GiftGrid } from './components'

const GiftExpertApp = () => {

  const [ categories, setCategories ] = useState([])

  const onAddCategory = ( newCategory ) => {
    if ( categories.includes( newCategory )) return 
    setCategories([ newCategory, ...categories ])
  }

  return (
    <>
    <h1>Holi</h1>

    <AddCategory 
      onAddCategory={ onAddCategory }
    />
  
    { categories.map( category => 
      <GiftGrid 
        category={ category } 
        key={ category }
      /> 
    )}
    
    </>
  )
}

export { GiftExpertApp }