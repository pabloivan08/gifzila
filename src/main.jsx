import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import { GiftExpertApp } from './GiftExperApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GiftExpertApp />
  </React.StrictMode>,
)