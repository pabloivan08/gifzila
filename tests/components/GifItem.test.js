import { render } from '@testing-library/react'
import { GifItem } from '../../src/components'

describe('Pruebas en <GifItem />', () => {
  const title = 'Goku'
  const url = 'https://dragonball.com/goku.jpg'
  
  test('Should match with the snapshot', () => {
    const { container } = render( <GifItem title={ title } url={ url }/> )
    expect{ container }.toMatchSnapshot()
  })
})