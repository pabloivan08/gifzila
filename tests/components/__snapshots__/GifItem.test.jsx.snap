// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Should match with the snapshot 1`] = `
<div>
  <div>
    <img
      alt="Goku"
      src="https://dragonball.com/goku.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
